pkg/lorry/plugin/proto/db_plugin.proto

```diff
@@ -0,0 +1,256 @@
+  string primary_fqdn = 2;
+
+  // The port on which the primary DB service listens.
+  string primary_port = 3;
+
+  // The Names of all members.
+  repeated string members = 4;
+
+  // The name of the new member's Pod.
+  string new_member = 5;
+
+  // The IP of the new member's Pod.
+  string new_member_ip = 6;
+
+  // Common metadata property for extention
+  map<string, string> metadata = 7;
+}
+
+message JoinMemberResponse {
+  // Common metadata property for extention
+  map<string, string> metadata = 1;
+}
+
+message LeaveMemberRequest {
+  DBInfo db_info = 1;
+
+  // The pod FQDN of the primary replica to check the role.
+  string primary_fqdn = 2;
+
+  // The port on which the primary DB service listens.
+  string primary_port = 3;
+
+  // The Names of all members.
+  repeated string members = 4;
+
+  // The name of the leave member's Pod.
+  string leave_member = 5;
+
+  // The IP of the leave member's Pod.
+  string leave_member_ip = 6;
+
+  // Common metadata property for extention
+  map<string, string> metadata = 7;
+}
+
+message LeaveMemberResponse {
+  // Common metadata property for extention
+  map<string, string> metadata = 1;
+}
+
+message ReadonlyRequest {
+  DBInfo db_info = 1;
+
+  // The reason for setting the replica as read-only.
+  string Reason = 2;
+
+  // Common metadata property for extention
+  map<string, string> metadata = 3;
+}
+
+message ReadonlyResponse {
+  // Common metadata property for extention
+  map<string, string> metadata = 1;
+}
+
+message ReadwriteRequest {
+  DBInfo db_info = 1;
+
+  // Common metadata property for extention
+  map<string, string> metadata = 2;
+}
+
+message ReadwriteResponse {
+  // Common metadata property for extention
+  map<string, string> metadata = 1;
+}
+
+message AccountProvisionRequest {
+  DBInfo db_info = 1;
+
+  // the user name of the new account.
+  string user_name = 2;
+
+  // the password of the new account.
+  string password = 3 [(kb_secret) = true];
+
+  // the role of the new account.
+  string role = 4;
+
+  // Common metadata property for extention
+}
+
+message AccountProvisionResponse {
+  // Common metadata property for extention
+  map<string, string> metadata = 1;
+}
+
+message SwitchoverRequest {
+  DBInfo db_info = 1;
+
+  // The pod FQDN of the primary replica.
+  string primary_fqdn = 2;
+
+  // The port on which the primary DB service listens.
+  string primary_port = 3;
+
+  // The Name of the old primary replica to demote.
+  string primary = 4;
+
+  // The Name of the new primary replica.
+  string candidate = 5;
+
+  // Force the switchover to happen even if the primary is not ready.
+  bool force = 6;
+
+  // Common metadata property for extention
+  map<string, string> metadata = 7;
+}
+
+message SwitchoverResponse {
+  // Common metadata property for extention
+  map<string, string> metadata = 1;
+}
+
+// DBPlugin service provides APIs to kubeblocks operator to exec component operation.
+service DBPlugin {
```

Comment:
What is the relationship with the original http protocol?

Reply:
The gRPC plugin is one of the implementations of the action handler, similar to the exec handler. The HTTP protocol is used for API communication between the controller and Lorry. An HTTP API call can invoke one or more gRPC plugin APIs.

Reply:
If the protocol is used for the communication with gRPC handers which are implemented by the provider, so:
1. why there is a gRPC server in lorry? And where is the protocol conversion from HTTP to gRPC, or the calling to the gRPC client?
2. the some problem in another PR(#7343), how does the provider know to implement this protocol?

Reply:
The gRPC server in Lorry was initially provided in early versions solely for RSM role probing purposes. It is currently maintained for compatibility reasons and it is planned to be removed in future versions.
As the same problem with other action handlers, it is necessary to document how to implement different types of action handlers and provide examples at action level.

Reply:
So where is the calling to the customized gRPC services?

Reply:
> The gRPC server in Lorry was initially provided in early versions solely for RSM role probing purposes. It is currently maintained for compatibility reasons and it is planned to be removed in future versions. As the same problem with other action handlers, it is necessary to document how to implement different types of action handlers and provide examples at action level.

And seems that the gRPC server and probe protocol are irrelevant to gRPC handlers, it's better not to involve their changes in this PR.